### DISCUSSION TITLE
Does this package support different connection names?

### DISCUSSION THREAD

**Discussion by ngunyimacharia** [richan-fongdasen/turso-laravel#7] [Q&A]
I'm working with https://tenancyforlaravel.com/ to develop multi-tenancy into an application. The package dynamically changes the connection name during application runtime from turso to tenant.

It seems that this package will still try to obtain the connection configuration from the turso config array instead of the tenant config array.

Does this mean that the package doesn't support different connection names? Is there a simple way to add this functionality?

**Comment by ngunyimacharia** [comment]
Changing TursoLaravelServiceProvider register from:

```
        $this->app->scoped(TursoManager::class, function () {
            return new TursoManager(config('database.connections.turso', []));
        });
```


to

```
        $this->app->scoped(TursoManager::class, function () {
            $connection = config('database.default', 'turso');

            return new TursoManager(config("database.connections.$connection", []));
        });
```

fixed my exact issue, but I don't believe this change is enough for a Pull Request yet.

**Comment by richan-fongdasen** [comment]
Hi @ngunyimacharia 👋🏼  

Thank you for using this Laravel package. I hope your multi-tenancy application development is going well!

Your question helped me realize that this package doesn't yet support multiple Turso database connections within the same application. I think some design changes would be needed to achieve that.

I have a few ideas that might work, but I'll need some time to experiment and ensure any new features work well and meet expectations.

**Reply by ngunyimacharia** [reply]
Glad to hear you already have some ideas around this. Eager to help and test this out. Will use a private fork with my fix for now, thank you.
